src/App.tsx

```diff
@@ -1,26 +1,170 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateToDo,
+} from './api/todos'; // Ensure createTodo is imported
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { Header } from './components/Header/Header';
+import { ToDoList } from './components/ToDoList/ToDoList';
+import { Error } from './components/Error/Error';
+import { Footer } from './components/Footer/Footer';
+function getVisibleToDos(newTodos: Todo[], newStatus: Status) {
+  switch (newStatus) {
+    case Status.Active:
+      return newTodos.filter(todo => !todo.completed);
+
+    case Status.Completed:
+      return newTodos.filter(todo => todo.completed);
 
-const USER_ID = 0;
+    default:
+      return newTodos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (error) {
+      timeoutId = setTimeout(() => setError(''), 3000);
+    }
+
+    return () => clearTimeout(timeoutId);
+  }, [error]);
```

Comment:
you dont need this use effect, you can remove error on finally when geting todos